src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -270,10 +270,10 @@ export class ExtHostTerminal extends Disposable {
 
 	public setShellType(shellType: TerminalShellType | number | undefined): boolean {
 		let extHostType: vscode.TerminalShellType | undefined;
-		switch (shellType) { // Following same order from proposed.terminalShellType.d.ts
-			case PosixShellType.Bash: extHostType = 1;
+		switch (shellType) {
+			case PosixShellType.Sh: extHostType = 1;
 			case PosixShellType.Fish: extHostType = 2;
-			case PosixShellType.Sh: extHostType = 3;
+			case PosixShellType.Bash: extHostType = 3;
```

Comment:
These are a little out of order now, bash is before fish in the .d.ts I see

Reply:
very nice catch! Now resolved [here](https://github.com/microsoft/vscode/pull/237624/commits/ce264daa7473b1d045b0bdeb6e0cb2a8f66af6fc)